verification/truststore/truststore.go

```diff
@@ -0,0 +1,126 @@
+// Package truststore reads certificates in a trust store
+package truststore
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	corex509 "github.com/notaryproject/notation-core-go/x509"
+	"github.com/notaryproject/notation-go/dir"
+)
+
+// Type is an enum for trust store types supported such as
+// "ca" and "signingAuthority"
+type Type string
+
+const (
+	TypeCA               Type = "ca"
+	TypeSigningAuthority Type = "signingAuthority"
+)
+
+var (
+	Types = []Type{
+		TypeCA,
+		TypeSigningAuthority,
+	}
+)
+
+// X509TrustStore provide list and get behaviors for the trust store
+type X509TrustStore interface {
+	// List named stores under storeType
+	// List(ctx context.Context, storeType Type) ([]string, error)
+
+	// GetCertificates returns certificates under storeType/namedStore
+	GetCertificates(ctx context.Context, storeType Type, namedStore string) ([]*x509.Certificate, error)
+}
+
+// NewX509TrustStore generates a new X509TrustStore
+func NewX509TrustStore(trustStorefs dir.SysFS) X509TrustStore {
+	return x509TrustStore{trustStorefs}
+}
+
+// x509TrustStore implements X509TrustStore
+type x509TrustStore struct {
+	trustStorefs dir.SysFS
+}
+
+// GetCertificates returns certificates under storeType/namedStore
+func (trustStore x509TrustStore) GetCertificates(ctx context.Context, storeType Type, namedStore string) ([]*x509.Certificate, error) {
+	if storeType == "" || namedStore == "" {
+		return nil, errors.New("storeType and namedStore cannot be empty")
+	}
+	path, err := trustStore.trustStorefs.SysPath(dir.X509TrustStoreDir(string(storeType), namedStore))
+	if err != nil {
+		return nil, err
+	}
+	// check path is valid
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return nil, fmt.Errorf("%q does not exist", path)
+	}
```

Comment:
There is a pitfall. See https://stefanxo.com/go-anti-patterns-os-isexisterr-os-isnotexisterr/